docs/hello_nextflow/02_hello_channels.md

```diff
@@ -0,0 +1,880 @@
+# Part 2: Hello Channels
+
+In Part 1 of this course (Hello World), we showed you how to provide a variable input to a process by providing the input in the process call directly: `sayHello(params.greet)`.
+That was a deliberately simplified approach.
+In practice, that approach has major limitations; namely that it only works for very simple cases where we only want to run the process once, on a single value.
+In most realistic workflow use cases, we want to process multiple values (experimental data for multiple samples, for example), so we need a more sophisticated way to handle inputs.
+
+That is what Nextflow **channels** are for.
+Channels are queues designed to handle inputs efficiently and shuttle them from one step to another in multi-step workflows, while providing built-in parallelism and many additional benefits.
+
+In this part of the course, you will learn how to use a channel to handle multiple inputs from a variety of different sources.
+You will also learn to use **operators** to transform channel contents as needed.
+
+_For training on using channels to connect steps in a multi-step workflow, see Part 3 of this course._
+
+---
+
+## 0. Warmup: Run `hello-channels.nf`
+
+We're going to use the workflow script `hello-channels.nf` as a starting point.
+It is equivalent to the script produced by working through Part 1 of this training course.
+
+Just to make sure everything is working, run the script once before making any changes:
+
+```bash
+nextflow run hello-channels.nf --greeting 'Hello Channels!'
+```
+
+```console title="Output"
+ N E X T F L O W   ~  version 24.10.0
+
+Launching `hello-channels.nf` [insane_lichterman] DSL2 - revision: c33d41f479
+
+executor >  local (1)
+[86/9efa08] sayHello | 1 of 1 ✔
+```
+
+As previously, you will find the output file named `output.txt` in the `results` directory (specified by the `publishDir` directive).
+
+```console title="output.txt" linenums="1"
+Hello Channels!
+```
+
+If that worked for you, you're ready to learn about channels.
+
+---
+
+## 1. Provide variable inputs via a channel explicitly
+
+We are going to create a **channel** to pass the variable input to the `sayHello()` process instead of relying on the implicit handling, which has certain limitations.
+
+### 1.1. Create an input channel
+
+There are a variety of **channel factories** that we can use to set up a channel.
+To keep things simple for now, we are going to use the most basic channel factory, called `Channel.of`, which will create a channel containing a single value.
+Functionally this will be exactly equivalent to how we had it set up before, but explicit instead of implicit.
```

Comment:
`Channel.value(` would be functionally/exactly equivalent to how we had it set up before, but explicit instead of implicit. The implicit conversion to channel creates a value channel. `Channel.of`, which we're using now, creates a queue channel.

I'd rephrase this phrase as something like "Instead of having Nextflow create a channel implicitly, we are doing this explicitly now."

Reply:
fair point, addressed in commit